source/Swagger/Decode.elm

```diff
@@ -68,7 +70,11 @@ decodeTypeByType ( type_, ref ) =
                     decodeArray
 
                 _ ->
-                    lazy (\_ -> decodeObject)
+                    case (properties, additionalProperties) of
+                        (Nothing, Just _) ->
+                            decodeDict
+                        _ ->
+                            lazy (\_ -> decodeObject)
```

Comment:
And this would instead become:

```elm
objectOrDict : ( List String, Maybe (List ( String, Type )), Maybe Type ) -> Type
objectOrDict ( required, properties, additionalProperties ) =
    case ( properties, additionalProperties ) of
        ( Nothing, Just addProps ) ->
            Dict_ addProps

        ( Just props, _ ) ->
            Object_ <| Properties <| List.map (property required) props

        _ ->
            Object_ <| Properties []
```